### PR TITLE
[plugin.video.jlss] v0.0.1

### PR DESCRIPTION
**Pull request by techkid6**
### Description
The Jimmy Lloyd Songwriter Showcase is a channel focusing on interviews with upcoming songwriters, presented like a linear television channel

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

### CODE REVIEW

**Comment by techkid6** [comment]
Oh no, I just messed this up again, I think.  That commit wasn't supposed to be on the PR.  If that's an issue, please let me know

**Comment by enen92** [comment]
yes that commit should not go with your PR

**Comment by techkid6** [comment]
I'll revert, no worries.

File: plugin.video.jlss/addon.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon
+  id="plugin.video.jlss"
+  name="The Jimmy Lloyd Songwriter Showcase"
+  version="0.0.1"
+  provider-name="Lab Rat Media">
+    <requires>
+        <import addon="xbmc.python" version="2.20.0"/>
```
Comment (enen92) [inline plugin.video.jlss/addon.xml]: 2.25.0 please (to match krypton)

File: plugin.video.jlss/addon.xml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        <import addon="script.module.feedparser" version="5.1.3"/>
+    </requires>
+    <extension point="xbmc.python.pluginsource" library="addon.py">
+        <provides>video</provides>
+    </extension>
+    <extension point="xbmc.addon.metadata">
+        <summary lang="en">The Jimmy Lloyd Songwriter Showcase</summary>
```
Comment (enen92) [inline plugin.video.jlss/addon.xml]: `en_GB` (new default since krypton)

File: plugin.video.jlss/addon.xml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        <description lang="en">Home to some of the best up and coming songwriters and bands, The Jimmy Lloyd Songwriter Showcase lets you get up close and personal with some of the most exciting talent around today.</description>
+        <disclaimer lang="en"></disclaimer>
+        <language>en</language>
```
Comment (enen92) [inline plugin.video.jlss/addon.xml]: Leave this one as `en`

File: plugin.video.jlss/changelog.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+v0.0.1 - Initial version
```
Comment (enen92) [inline plugin.video.jlss/changelog.txt]: Changelog is deprecated, you can set a `<news></news>` element within addon.xml. See: https://kodi.wiki/view/addon.xml#.3Cnews.3E

File: plugin.video.jlss/addon.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    url = entry["media_content"][0]["url"]
+    thumbnail = entry["media_thumbnail"][0]["url"]
+    info = {}
+    info["title"] = entry["title"]
+    info["plot"] = entry["summary"]
+    info["genre"] = entry["category"]
+    info["duration"] = int(entry["media_content"][0]["duration"])
+    info["mediatype"] = "episode"
+    
+    listitem = xbmcgui.ListItem(info["title"], thumbnailImage=thumbnail)
```
Comment (enen92) [inline plugin.video.jlss/addon.py]: thumbnailImage kwarg is deprecated please use `listitem.setArt()`: https://codedocs.xyz/xbmc/xbmc/group__python__xbmcgui__listitem.html#gad3f9b9befa5f3d2f4683f9957264dbbe

**Comment by enen92** [comment]
@techkid6 I am a bit reluctant to accept this as a plugin. I don't want to sound picky but a plugin is in fact an addon that manipulates Kodi's virtual filesystem. It creates new directories and new playable items within those directories. When kodi is pointed to a `plugin://` path and if the path `isPlayable`, Kodi will call the plugin and expect it to `setResolvedUrl` on the item for playback.
You addon is more a script than a plugin (https://forum.kodi.tv/showthread.php?tid=48954&pid=316937). Note that being a script does not require that your addon implements a new window. To be honest, plugins should not use xbmc.Player() directly!

What I would suggest:
1) turn your addon into a script instead of a plugin
- modify its id to script.video.jlss in addon.xml
- modify `<extension point="xbmc.python.pluginsource"` to `<extension point="xbmc.python.script"`

Keep the `<provides>video</provides>` so that the addon still shows up in the video addons section of the addon browser

2) While you are at it make the entrypoint of your script as minimal as possible. Python entrypoints will not get compiled to bytecode by Kodi's python interpreter. So the entrypoint referred in the addon.xml should just import a module and run it (while the main script code is kept in the imported module). Example:

```python
from resources.lib import addon
addon.run()
```

in addon.py stored in the resources/lib folder:
```python

def main():
    playlist = xbmc.PlayList(1)
    playlist.clear()
    for item in get_all_videos(PLAYLIST_URL):
        url, listitem = item
        playlist.add(url, listitem)
        xbmc.log(str(url))
    playlist.shuffle()

    xbmc.Player().play(playlist) 
```

Note this is just a recomendation

3) Close this pull request and open another in `repo-scripts` in the krypton branch:
https://github.com/xbmc/repo-scripts/


Again, I don't want to sound picky. Thanks a lot

**Comment by techkid6** [comment]
No worries at all, this is all really good information to know!  While I'm at it, should I also consider migrating `plugin.video.myjamtv` to a script addon as well?

I'm going to make these changes as soon as I have a chance

**Comment by enen92** [comment]
I'd say you don't need to make the other addon a script but maybe make some changes to implement the playback properly (making it what a plugin is supposed to be). See this example for a recent PR:

https://github.com/Lunatixz/repo-plugins/blob/454f6f155c491bbd5d72b5bd44a69370fed26313/plugin.video.playonbrowser/default.py#L246

**Pull request closed without merging**